include/avt_vimba_camera/status_camera.h

```diff
@@ -0,0 +1,27 @@
+#include <ros/ros.h>
+#include <avt_vimba_camera/mono_camera.h>
+#include <cav_msgs/SystemAlert.h>
+#include <cav_msgs/DriverStatus.h>
+#include <ros/callback_queue.h>
+#include <boost/thread.hpp>
+
+class StatusCamera
+{
+    cav_msgs::DriverStatus status_;
+    ros::Publisher status_pub_ ;
+    ros::Subscriber alert_sub_;
+    ros::NodeHandle nh_;
+    ros::NodeHandle nhp_;
+    boost::thread* cam_thread_;
```

Comment:
Why are we using a pointer to the thread?

Reply:
Portable thread-local storage with boost::thread_specific_ptr
boost::thread_specific_ptr provides a portable mechanism for thread-local storage that works on all compilers supported by Boost.Thread. Each instance of boost::thread_specific_ptr represents a pointer to an object (such as errno) where each thread must have a distinct value. The value for the current thread can be obtained using the get() member function, or by using the * and -> pointer deference operators. Initially the pointer has a value of NULL in each thread, but the value for the current thread can be set using the reset() member function.

If the value of the pointer for the current thread is changed using reset(), then the previous value is destroyed by calling the cleanup routine. Alternatively, the stored value can be reset to NULL and the prior value returned by calling the release() member function, allowing the application to take back responsibility for destroying the object.

Reply:
You copy & paste a paragraph from boost:thread_specific_ptr documentation. Can you please explain how is that related to me question? Correct me if I am wrong. We are talking about boost::thread which has nothing to do with thread local storage.

Reply:
Because thread type itself is a pointer.